File: Differential Equations - Part I/Codes/Harmonic1.py
import math as m
def g(theta0,omega0):
	return omega0
def f(theta0,omega0,t):
	return -w0**2.*theta0
gamma=.2
w0=1.
omega=1.
theta=0.
h=0.1
N=1000
delta=1.
w=.2
F=2.
for i in range(N):
	print i*h,theta,omega	
	theta1=theta+h*g(theta,omega)
	omega1=omega+h*f(theta,omega,i*h)
	theta=theta1	
	omega=omega1

File: Differential Equations - Part I/Codes/Harmonic1.cpp
#include<iostream>
#include<cmath>
const double w0=1.;
double g(double theta0,double omega0,double t);
double f(double theta0,double omega0,double t);
int main(void){
  double omega=1;
  double theta=0;
  double h=0.1;
  int N=1000;
  for(int i=0;i<N;i++){
    std::cout<<i*h<<'\t'<<theta<<'\t'<<omega<<std::endl;	
    omega=omega+h*f(theta,omega,i*h);
    theta=theta+h*g(theta,omega,i*h);
  }
  return 0;
}
double g(double theta0,double omega0,double t){
  return omega0;}
double f(double theta0,double omega0,double t){
  return -w0*w0*theta0;}
